platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/sources/process/ProcessOutputValueSource.java

```diff
@@ -122,7 +122,7 @@ public ExecOutputData obtain() {
 
         ExecResult r = execOperations.exec(spec -> {
             spec.commandLine(getParameters().getCommandLine().get());
-            spec.setIgnoreExitValue(getParameters().getIgnoreExitValue().getOrElse(false));
+            spec.getIgnoreExitValue().set(getParameters().getIgnoreExitValue().orElse(false));
```

Comment:
❓ Do we need the `orElse(false)` here?

Reply:
I don't think we need it, since spec.getIgnoreExitValue() has convention set to `false`

Reply:
Actually it looks like we need it. If we pass empty provider to `getIgnoreExitValue()` convention is not used and `.get()` later in action can fail

Reply:
Is this how convention _should_ work here?

Taking a step back and looking at this from a UX point of view: I feel like this should "just work." Having to add `orElse(false)` goes strongly against that.

Reply:
According to docs, this how `convention` works (yes it feels wrong):
https://github.com/gradle/gradle/blob/99560e13819684554dbab45fee5aa7f826c24e97/subprojects/core-api/src/main/java/org/gradle/api/provider/Property.java#L71C1-L72C74